website/project/model.py

```diff
@@ -1573,9 +1582,35 @@ def save(self, *args, **kwargs):
         if settings.PIWIK_HOST and update_piwik:
             piwik_tasks.update_node(self._id, saved_fields)
 
+        # Ensure Mailing list subscription is set up
+        if first_save:
+            self.get_or_create_mailing_list_subscription()
+
         # Return expected value for StoredObject::save
         return saved_fields
 
+    def get_or_create_mailing_list_subscription(self):
+        if self.mailing_enabled:
+            # Avoid circular import
+            from website.models import NotificationSubscription
+            from website.notifications.utils import to_subscription_key
+            try:
+                subscription = NotificationSubscription(
+                    _id=to_subscription_key(self._id, 'mailing_list_events'),
+                    owner=self,
+                    event_name='mailing_list_events'
+                )
+                subscription.save()
+            except KeyExistsException:
+                subscription = NotificationSubscription.load(to_subscription_key(self._id, 'mailing_list_events'))
+            else:
+                subscription.add_user_to_subscription(self.creator, 'email_transactional', save=True)
```

Comment:
`else` is fired if the try block succeeds- do we really need to be saving the `subscription` twice in a row?